### PR TITLE
Add driver type to nodePool resources

### PR DESCRIPTION
**Pull request by cbron**
Issue: https://github.com/rancher/rancher/issues/22080

Types PR: https://github.com/rancher/types/pull/971

Adds a virtual field (not stored in db) to the nodepool resource via formatter that identifies the driver type (string). 

Hit `/v3/schemas/nodePool` to see schema
Hit `/v3/nodePool` to see resources

Schema
driver-on-schema-2

Resource
driver-on-object

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,6 +7,7 @@ replace (
 	github.com/knative/pkg => github.com/rancher/pkg v0.0.0-20190514055449-b30ab9de040e
 	github.com/matryer/moq => github.com/rancher/moq v0.0.0-20190404221404-ee5226d43009
 	k8s.io/client-go => github.com/rancher/client-go v1.14.5-rancher.1
+	github.com/rancher/types => github.com/cbron/types v0.0.0-20190830194953-8ea480efd1fe
```
Comment (cbron) [inline go.mod]: will revert

File: vendor/github.com/rancher/types/apis/management.cattle.io/v3/schema/schema.go [hunk 2]
```diff
@@ -341,13 +341,23 @@ func nodeTypes(schemas *types.Schemas) *types.Schemas {
...
+				Update: false,
+			}
+		}).
 		MustImport(&Version, v3.NodeDrainInput{}).
 		MustImportAndCustomize(&Version, v3.Node{}, func(schema *types.Schema) {
 			labelField := schema.ResourceFields["labels"]
 			labelField.Create = true
 			labelField.Update = true
 			schema.ResourceFields["labels"] = labelField
+			annotationField := schema.ResourceFields["annotations"]
```
Comment (cbron) [inline vendor/github.com/rancher/types/apis/management.cattle.io/v3/schema/schema.go]: unrelated

File: vendor/github.com/rancher/types/apis/management.cattle.io/v3/schema/schema.go [hunk 3]
```diff
@@ -341,13 +341,23 @@ func nodeTypes(schemas *types.Schemas) *types.Schemas {
 		AddMapperForType(&Version, v3.NodeDriver{}, m.DisplayName{}).
 		AddMapperForType(&Version, v3.NodeTemplate{}, m.DisplayName{}).
 		MustImport(&Version, v3.PublicEndpoint{}).
-		MustImport(&Version, v3.NodePool{}).
+		MustImportAndCustomize(&Version, v3.NodePool{}, func(schema *types.Schema) {
+			schema.ResourceFields["driver"] = types.Field{
+				Type: "string",
```
Comment (cbron) [inline vendor/github.com/rancher/types/apis/management.cattle.io/v3/schema/schema.go]: String because: https://github.com/rancher/types/blob/master/apis/management.cattle.io/v3/machine_types.go#L47

**Comment by dramich** [comment]
The change looks fine but this should be split into multiple commits, one for vendor and another for your changes.

File: tests/integration/suite/test_node.py [hunk 4]
```diff
@@ -12,7 +12,7 @@
 def test_node_fields(admin_mc):
     cclient = admin_mc.client
     fields = {
-        'annotations': 'r',
+        'annotations': 'cru',
```
Comment (cbron) [inline tests/integration/suite/test_node.py]: Fixes unrelated change

**Comment by dramich** [comment]
lgtm
